enclave-runtime/src/sidechain_impl.rs

```diff
@@ -0,0 +1,268 @@
+//! Implement sidechain traits that can't be implemented outside.
+//!
+//! Todo: Once we have put the `top_pool` stuff in an entirely different crate we can
+//! move most parts here to the sidechain crate.
```

Comment:
Note here: As the comment says, these parts exist in the enclave-runtime because they have dependencies on parts that live in the enclave-runtime and are not available outside (like the TOP pool and RPC author).